apps/src/turtle/customLevelBlocks.js

```diff
@@ -39,28 +39,80 @@ function createACircleCode (size, gensym, indent) {
     indent + '}\n'].join('\n');
 }
 
-
 /**
- * Same as draw_a_square, except inputs are not inlined
+ * Returns an initialization object that sets up blockly attributes;
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @param [argument] - Optional parameter for blocks that accept a value
+ *    argument. This is the title of the argument.
+ * @param [titleValue] - Optional parameter to associate a value with the
+ *    title.
+ * @return the initialization object
  */
-function installDrawASquare(blockly, generator, gensym) {
-  // Create a fake "draw a square" function so it can be made available to users
-  // without being shown in the workspace.
-  blockly.Blocks.draw_a_square_custom = {
-    // Draw a square.
+function makeInitializer(title, argument, titleValue) {
+  return {
     init: function() {
       this.setHSV(94, 0.84, 0.60);
-      this.appendDummyInput()
-          .appendTitle(msg.drawASquare());
-      this.appendValueInput('VALUE')
-          .setAlign(blockly.ALIGN_RIGHT)
-          .setCheck(Blockly.BlockValueType.NUMBER)
-              .appendTitle(msg.lengthParameter() + ':');
+
+      this.appendDummyInput().appendTitle(title, titleValue);
+
+      if (argument !== undefined) {
+        this.appendValueInput('VALUE')
+            .setAlign(Blockly.ALIGN_RIGHT)
+            .setCheck(Blockly.BlockValueType.NUMBER)
+            .appendTitle(argument + ':');
+      }
+
       this.setPreviousStatement(true);
       this.setNextStatement(true);
       this.setTooltip('');
     }
   };
+}
+
+/**
+ * Returns an initialization object that sets up blockly attributes on blocks
+ * that associate a value with their titles.
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @param value - The value to associate with the title.
+ * @return the initialization object
+ */
+function makeInitializerWithAssociatedTitleValue(title, value) {
+  return makeInitializer(title, undefined, value);
+}
+
+/**
+ * Returns an initialization object that sets up blockly attributes on blocks
+ * that take a size parameter.
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @return the initialization object
+ */
+function makeInitializerWithSizeParam(title) {
+  return makeInitializer(title, msg.sizeParameter());
+}
```

Comment:
Yea, good observation. I simply enumerated every variant of block initialization that existed in customLevelBlocks. Thought it might be useful to someone new coming into the file to see them all in one place.

Looking at the file history, folks generally just copy/paste some other implementation. The theory was that these would serve as a reference so someone can look and say "Ah, this is the variant that I need."

I suppose most people coming into the file already have a block in mind that is similar to what they want anyway (e.g. I need a block that takes similar parameters to a snowman), so they can just search for that and copy the impl.

These are removed now.